cupy/cuda/compiler.py

```diff
@@ -25,18 +25,25 @@ class NVCCException(Exception):
     pass
 
 
-def _run_nvcc(cmd, cwd):
+def _run_nvcc(cmd, cwd, log_stream=None):
     try:
-        return subprocess.check_output(cmd, cwd=cwd, stderr=subprocess.STDOUT)
+        stdout = subprocess.check_output(cmd, cwd=cwd,
+                                         stderr=subprocess.STDOUT,
+                                         universal_newlines=True)
+        if log_stream == 'stdout':
```

Comment:
The idea is to use `sys.stdout` value as the `log_stream` parameter so we can avoid explicitly calling `print` 😃

Reply:
@emcastillo Are you saying to replace `print` with

```python
sys.stdout = stdout
```

Reply:
no, what I mean is to pass `sys.stdout` as the `log_stream` for the kernel.

```python
RawKernel(*args, log_stream=sys.stdout)
```
So inside this code we only do
```
log_stream.write(log)
```
The user should be responsible for opening the stream

Reply:
@emcastillo Check it out now. I think I understand what you're saying, and yes, it is much simpler. :smile: